torch/csrc/jit/interpreter.cpp

```diff
@@ -58,12 +58,79 @@ namespace torch { namespace jit {
 void desugarTripCounts(Block * b) {
   for(auto n : b->nodes()) {
     if(n->kind() == kLoop) {
-
-      // remove the trip count from Loop inputs, we don't support it yet
+      // First, replace the max trip count input with the initial trip count.
+      // This means we can treat trip count as a loop-carried dependency and
+      // simply emit an increment in the body of the loop
+      Value* max_trip_count_value = n->input(0);
+      Value* initial_trip_count =
+          n->owningGraph()
+              ->createConstant(at::zeros(at::CPU(at::kInt), {1}))
+              ->output();
+      initial_trip_count->node()->insertBefore(n);
+      // Also move trip count to the end because we need to mutate it after this
+      // transformation and it's easier to append an output in the block than it
+      // is to prepend.
       n->removeInput(0);
-      JIT_ASSERT(n->blocks()[0]->inputs()[0]->uses().size() == 0 &&
-        "NYI - use of trip count variable");
-      n->blocks()[0]->eraseInput(0);
+      n->addInput(initial_trip_count);
+
+      auto body_block = n->blocks()[0];
+
+      Value* block_trip_count_input = body_block->addInput();
+      body_block->inputs()[0]->replaceAllUsesWith(block_trip_count_input);
+      body_block->eraseInput(0);
+      n->addOutput();
+
+      auto insertAfter =
```

Comment:
Is there any reason why we can't use `WithInsertPoint` + `create` (you can give list of inputs to `create` too IIRC)? That's what we're using everywhere else and it seems to result in much simpler code